RadClass/RadClass.py

```diff
@@ -171,17 +195,17 @@ def iterate(self):
         Only runs for a set node (datapath) with data already queued.
         '''
         bar = progressbar.ProgressBar(max_value=100, redirect_stdout=True)
-        inverse_dt = 1.0 / (self.processor.timestamps[-1]
-                            - self.processor.timestamps[0])
+        inverse_dt = 1.0 / (self.stop_i - self.start_i)
+        offset = self.start_i
 
         log_interval = 10000  # number of samples analyzed between log updates
```

Comment:
non-blocking: Doesn't it make sense to define this by the start and stop times now? So that you get a minimum/maximum number of updates in any given interval?

Reply:
I'm not sure I understand but I offered a change in the most recent commit. If you mean to change `log_interval` to be dependent on start/stop times, I am having a hard time visualizing how you could change the frequency of updates in this way.

Reply:
Something like `log_interval = max(min((stop_i - start_)/100, 10000),10)`

This will update 100 times, but wait no more than 10,000 s between updates and no fewer than 10